deploy/modules/gcp/tidb-cluster/main.tf

```diff
@@ -0,0 +1,159 @@
+resource "google_container_node_pool" "pd_pool" {
+  // The monitor pool is where tiller must first be deployed to.
+  depends_on = [google_container_node_pool.monitor_pool]
+  provider   = google-beta
+  project    = var.gcp_project
+  cluster    = var.gke_cluster_name
+  location   = var.gke_cluster_location
+  name       = "${var.cluster_name}-pd-pool"
+  node_count = var.pd_node_count
+
+  management {
+    auto_repair  = false
+    auto_upgrade = false
+  }
+
+  node_config {
+    machine_type    = var.pd_instance_type
+    local_ssd_count = 0
+
+    taint {
+      effect = "NO_SCHEDULE"
+      key    = "dedicated"
+      value  = "${var.cluster_name}-pd"
+    }
+
+    labels = {
+      dedicated = "${var.cluster_name}-pd"
+    }
+
+    tags         = ["pd"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+  }
+}
+
+resource "google_container_node_pool" "tikv_pool" {
+  provider   = google-beta
+  project    = var.gcp_project
+  cluster    = var.gke_cluster_name
+  location   = var.gke_cluster_location
+  name       = "${var.cluster_name}-tikv-pool"
+  node_count = var.tikv_node_count
+
+  management {
+    auto_repair  = false
+    auto_upgrade = false
+  }
+
+  node_config {
+    machine_type = var.tikv_instance_type
+    image_type   = "UBUNTU"
+    // This value cannot be changed (instead a new node pool is needed)
+    // 1 SSD is 375 GiB
+    local_ssd_count = 1
+
+    taint {
+      effect = "NO_SCHEDULE"
+      key    = "dedicated"
+      value  = "${var.cluster_name}-tikv"
+    }
+
+    labels = {
+      dedicated = "${var.cluster_name}-tikv"
+    }
+
+    tags         = ["tikv"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+  }
+}
+
+resource "google_container_node_pool" "tidb_pool" {
+  // The pool order is tikv -> monitor -> pd -> tidb
+  depends_on = [google_container_node_pool.pd_pool]
+  provider   = google-beta
+  project    = var.gcp_project
+  cluster    = var.gke_cluster_name
+  location   = var.gke_cluster_location
+  name       = "${var.cluster_name}-tidb-pool"
+  node_count = var.tidb_node_count
+
+  management {
+    auto_repair  = false
```

Comment:
Again, I would not enable `auto_upgrade` because it can reduce the service availability, and like any upgrade has the potential to introduce bugs into a system that was already in a known state.